### PR TITLE
Add confirmation dialog for Delete button in Details View

### LINKED ISSUE #107: User is not warned when deleting a job or job definition in Details view
## Description

When the user clicks "Delete Job" or "Delete Job Definition" from the details view, the job or job definition is deleted immediately, with no confirmation step. This is irreversible.

## Expected behavior

The user should see a confirmation dialog (e.g., a modal) when they click a delete button so that they can confirm their decision to delete.

### PR DESCRIPTION
**Pull request by andrii-i**
Closes #107

Preview:
final_delete_confirmation

### CODE REVIEW

File: src/components/delete-with-confirmation-button.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        )}
+        <DialogActions>
+          <Button
+            variant="outlined"
+            onClick={_ => {
+              handleClose();
+              props.handleDelete();
+            }}
+          >
+            Yes
```
Comment (jweill-aws) [inline src/components/delete-with-confirmation-button.tsx:43]: JupyterLab tends to use the [Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/components/presentation/alerts/): the primary button is on the right, and "Yes" and "No" are discouraged in favor of action verbs. In this case, the primary button should be "Delete" (localized) and the secondary button should be "Cancel".

File: src/mainviews/detail-view/job-definition.tsx [hunk 2]
```diff
@@ -61,13 +62,11 @@ export function JobDefinition(props: IJobDefinitionProps): JSX.Element {
...
-        variant="contained"
-        color="error"
-        onClick={handleDeleteJobDefinition}
-      >
-        {trans.__('Delete Job Definition')}
-      </Button>
+      <DeleteWithConfirmationButton
+        handleDelete={handleDeleteJobDefinition}
+        title={trans.__('Delete Job Definition')}
+        text={trans.__('Are you sure you want to delete this Job Definition?')}
```
Comment (jweill-aws) [inline src/mainviews/detail-view/job-definition.tsx:68]: "Job" and "job definition" are not proper nouns, and so should only be capitalized at the start of a sentence.

```suggestion
        text={trans.__('Are you sure that you want to delete this job definition?')}
```

File: src/mainviews/detail-view/job-detail.tsx [hunk 3]
```diff
@@ -101,9 +102,11 @@ export function JobDetail(props: IJobDetailProps): JSX.Element {
       <Button variant="outlined" onClick={handleRerunJob}>
         {trans.__('Rerun Job')}
       </Button>
-      <Button variant="contained" color="error" onClick={handleDeleteJob}>
-        {trans.__('Delete Job')}
-      </Button>
+      <DeleteWithConfirmationButton
+        handleDelete={handleDeleteJob}
+        title={trans.__('Delete Job')}
+        text={trans.__('Are you sure you want to delete this Job?')}
```
Comment (jweill-aws) [inline src/mainviews/detail-view/job-detail.tsx:108]:
```suggestion
        text={trans.__('Are you sure that you want to delete this job?')}
```

File: src/mainviews/detail-view/job-definition.tsx (hunk 2 above)
Comment (andrii-i) [reply src/mainviews/detail-view/job-definition.tsx:68]: I've been capitalizing job and job definition every time for the past 1.5 months 💀

File: src/components/delete-with-confirmation-button.tsx (hunk 1 above)
Comment (andrii-i) [reply src/components/delete-with-confirmation-button.tsx]: Updated:
Screen Shot 2022-10-13 at 4 55 30 PM

**Comment by andrii-i** [comment]
Updated preview:
Screen Shot 2022-10-13 at 4 55 30 PM

File: src/components/delete-with-confirmation-button.tsx (hunk 1 above)
Comment (jweill-aws) [reply src/components/delete-with-confirmation-button.tsx]: The primary button is typically colored in in our dialogs; let's make the delete button red, like in this example dialog. Also note that the selected button is "Cancel," so if the user hits <kbd>SPACE</kbd> or <kbd>ENTER</kbd>, they will take the non-destructive action.

image

Comment (andrii-i) [reply src/components/delete-with-confirmation-button.tsx]: Updated versions of the dialog:
Screen Shot 2022-10-13 at 5 58 47 PM
Screen Shot 2022-10-13 at 5 58 35 PM

**Comment by andrii-i** [comment]
Latest preview:
Screen Shot 2022-10-13 at 5 58 47 PM
Screen Shot 2022-10-13 at 5 58 35 PM

**Comment by andrii-i** [comment]
Fixed everything, added auto-focus on Cancel

**Review by jweill-aws — APPROVED** [review]

**Pull request merged**